File: app/src/main/java/com/testapp/testapp/model/rest/VenueApi.java
package com.testapp.testapp.model.rest;

import com.testapp.testapp.model.entity.response.Response;
import com.testapp.testapp.model.entity.response.ResponsePhotos;
import com.testapp.testapp.model.entity.response.ResponseTips;
import com.testapp.testapp.model.entity.response.ResponseVenue;
import com.testapp.testapp.model.entity.response.ResponseSearchVenues;

import java.util.Map;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.QueryMap;

/**
 * Created by troll on 03.11.2017.
 */

public interface VenueApi {

    @GET("venues/search")
    Call<Response<ResponseSearchVenues>> searchVenues(@QueryMap Map<String, String> params);

    @GET("venues/{venueId}")
    Call<Response<ResponseVenue>> venueDetails(@Path("venueId") String id, @QueryMap Map<String, String> params);

    @GET("venues/{venueId}/photos")
    Call<Response<ResponsePhotos>> venuePhotos(@Path("venueId") String id, @QueryMap Map<String, String> params);

    @GET("venues/{venueId}/tips")
    Call<Response<ResponseTips>> venueTips(@Path("venueId") String id, @QueryMap Map<String, String> params);
}

File: app/src/main/java/com/testapp/testapp/kotlin/model/rest/VenueApi.kt
package com.testapp.testapp.kotlin.model.rest

import com.testapp.testapp.kotlin.model.entity.response.*
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.QueryMap

interface VenueApi {

    @GET("venues/search")
    fun searchVenues(@QueryMap params: Map<String, String>) : Call<Response<ResponseSearchVenues>>

    @GET("venues/{venueId}")
    fun venueDetails(@Path("venueId") id: String, @QueryMap params: Map<String, String>) : Call<Response<ResponseVenue>>

    @GET("venues/{venueId}/photos")
    fun venuePhotos(@Path("venueId") id: String, @QueryMap params: Map<String, String>) : Call<Response<ResponsePhotos>>

    @GET("venues/{venueId}/tips")
    fun venueTips(@Path("venueId") id: String, @QueryMap params: Map<String, String>) : Call<Response<ResponseTips>>
}